datafusion/physical-expr/src/expressions/is_null.rs

```diff
@@ -74,9 +77,16 @@ impl PhysicalExpr for IsNullExpr {
     fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
         let arg = self.arg.evaluate(batch)?;
         match arg {
-            ColumnarValue::Array(array) => Ok(ColumnarValue::Array(Arc::new(
-                compute::is_null(array.as_ref())?,
-            ))),
+            ColumnarValue::Array(array) => {
+                let bool_array = if let Some(union_array) =
+                    array.as_any().downcast_ref::<UnionArray>()
+                {
+                    union_is_null(union_array)?
+                } else {
+                    compute::is_null(array.as_ref())?
+                };
+                Ok(ColumnarValue::Array(Arc::new(bool_array)))
+            }
```

Comment:
instead of replicating the special case for `UnionArray` here and in `is_not_null` what do you think about making a wrapper for `compute::is_null` in  DataFusion:

```rust
/// wraper around arrow::compute::is_null
fn is_null(datum: &Datum) -> Result<BooleanArray> {
  if let Some(union_array) =
                    array.as_any().downcast_ref::<UnionArray>()
                {
                    union_is_null(union_array)
                } else {
                    compute::is_null(array.as_ref())
                }
  }
```

The idea being that then when the fix is available in arrow-rs then we can simple remove the wrapper

Reply:
I've done what I think you mean here.

The only material change is that now for the "is not null" case we use effectively use `compute::not(compute::is_null(...))` instead of `compute::is_not_null`, I'm not sure if those will compile to the same thing, or if you care about any resultant differences.